docs/spec_draft.md

```diff
@@ -1406,6 +1407,61 @@ and produces a `result` tensor. More formally,
 ```
 
 [Back to Ops](#index-of-ops)
+
+## stablehlo.rng
+
+### Semantics
+
+Constructs a `result` tensor of a given shape `shape` with random numbers
+generated using the `rng_distribution` algorithm. If `rng_distribution` $=$
+`UNIFORM`, then the random numbers are generated following the uniform
+distribution over the interval [`a`, `b`). If `rng_distribution` $=$ `NORMAL`,
+then the random numbers are generated following the normal distribution with
+mean = `a` and standard deviation = `b`.
+
+If `rng_distribution = UNIFORM` and `b` $\le$ `a`, the result is
+implementation-defined.
+
+### Inputs
+
+| Name               | Type                                                 |
+|--------------------|------------------------------------------------------|
+| `a`                | constant of integer, boolean, or floating-point type |
+| `b`                | constant of integer, boolean, or floating-point type |
+| `shape`            | 1-dimensional tensor of integer type                 |
+| `rng_distribution` | enum of `UNIFORM` and `NORMAL`                       |
+
+### Outputs
+
+| Name     | Type                                               |
+|----------|----------------------------------------------------|
+| `result` | tensor of integer, boolean, or floating-point type |
+
+### Constraints
+
+  * (C1) `a`, `b`, and `result` have the same element type.
+  * (C2) dim(`result`, `i`) = `shape[i]` for all `i` $\in$ [0, size(`shape`)).
+  * (C3) If `rng_distribution = NORMAL`, `a`, `b`, and `result` have the same
+    floating-point element type.
```

Comment:
How about reordering C2 and C3, so that the related constraints C1 and C3 are located next to each other?